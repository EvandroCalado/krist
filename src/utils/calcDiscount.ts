export const calcDiscount = (value: number, discount: number) => {
  const valueWithDiscount = (value * (100 - discount)) / 100;

  if (discount === 0) return value;

  return valueWithDiscount.toFixed(2);
};
