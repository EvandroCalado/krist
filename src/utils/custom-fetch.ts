import axios from 'axios';

export const customFetch = axios.create({
  baseURL: import.meta.env.VITE_STRAPI_URL,
});
