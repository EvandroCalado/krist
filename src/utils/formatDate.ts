export const formatDate = (dateString: string): string => {
  const date = new Date(dateString);

  return date.toLocaleString('pt-BR', {
    year: 'numeric',
    month: 'long',
    day: 'numeric',
  });
};
