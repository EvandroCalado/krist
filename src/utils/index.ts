export * from './calcDiscount';
export * from './custom-fetch';
export * from './formatDate';
export * from './linksProfileMenu';
export * from './test-utils';
