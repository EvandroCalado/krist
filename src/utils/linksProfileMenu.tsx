import { Heart, MapPin, Package, Settings, UserRound } from 'lucide-react';

export const linksProfileMenu = [
  {
    id: 1,
    label: 'informações pessoais',
    path: 'profile',
    icon: <UserRound size={16} />,
  },
  {
    id: 2,
    label: 'meus pedidos',
    path: 'orders',
    icon: <Package size={16} />,
  },
  {
    id: 3,
    label: 'lista de desejos',
    path: 'wishlist',
    icon: <Heart size={16} />,
  },
  {
    id: 4,
    label: 'meus endereços',
    path: 'addresses',
    icon: <MapPin size={16} />,
  },
  {
    id: 5,
    label: 'configurar',
    path: 'settings',
    icon: <Settings size={16} />,
  },
];
