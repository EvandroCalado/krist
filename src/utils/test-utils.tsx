import { render, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';
import { ThemeProvider } from 'styled-components';

import theme from 'styles/theme';

type CustomRenderProps = Omit<RenderOptions, 'queries'>;

const customRender = (
  ui: ReactElement,
  { ...renderOptions }: CustomRenderProps = {},
) => render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, renderOptions);

export * from '@testing-library/react';
export { customRender as render };
