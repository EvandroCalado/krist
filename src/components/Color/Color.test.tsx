import { screen } from '@testing-library/react';

import { customRender } from 'utils/test-utils';

import { Color } from '.';

describe('Color', () => {
  it('should render default', () => {
    customRender(<Color color="preto" label="preto" colorHex="#000" />);

    const color = screen.getByTestId('color');
    const input = screen.getByTestId('checkbox');
    const label = screen.getByText('preto');

    expect(color).toBeInTheDocument();
    expect(color).toHaveStyle({
      'background-color': '#000',
    });
    expect(color).toHaveClass('active');

    expect(input).toHaveStyle({
      display: 'none',
      visibility: 'hidden',
    });

    expect(label).toBeInTheDocument();
    expect(label).toHaveTextContent('preto');
  });

  it('should render not active', () => {
    customRender(<Color color="preto" label="branco" colorHex="#000" />);

    const color = screen.getByTestId('color');

    expect(color).not.toHaveClass('active');
  });

  it('should render a snapshot', () => {
    const { container } = customRender(
      <Color color="preto" label="preto" colorHex="#000" />,
    );

    expect(container).toMatchSnapshot();
  });
});
