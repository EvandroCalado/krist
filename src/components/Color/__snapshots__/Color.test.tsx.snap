// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color should render a snapshot 1`] = `
.c0 {
  width: 100%;
  display: flex;
  align-items: center;
  gap: .8rem;
}

.c3 {
  width: 100%;
  background-color: transparent;
  outline: none;
  border: none;
  text-align: left;
  text-transform: capitalize;
  font-size: 1.2rem;
  padding: .8rem 1rem;
  border-radius: 0.5rem;
  cursor: pointer;
}

.c3:hover {
  background-color: #f0f0f0;
}

.c2 {
  display: none;
  visibility: hidden;
}

.c1 {
  width: 3.5rem;
  height: 3rem;
  background-color: #000;
  border-radius: 0.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  border: 1px solid #f0f0f0;
  margin: 0.3rem;
}

.c1.active {
  outline: 2px solid #131118;
  outline-offset: 1px;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1 active"
      data-testid="color"
    />
    <input
      class="c2"
      data-testid="checkbox"
      id="preto"
      type="checkbox"
    />
    <label
      class="c3"
      for="preto"
    >
      preto
    </label>
  </div>
  ,
</div>
`;
